server/config.go

```diff
@@ -45,6 +45,13 @@ func (cfg *Config) Check() error {
 		}
 	}
 
+	// provide dummy values to eigenda client config. Since the client won't be called in this
+	// mode it doesn't matter.
+	if cfg.MemstoreEnabled {
+		cfg.EdaClientConfig.SvcManagerAddr = "0x0000000000000000000000000000000000000000"
+		cfg.EdaClientConfig.EthRpcUrl = "http://0.0.0.0:666"
+	}
```

Comment:
This works well for when memstore is enabled, but the issue now is that eth_rpc and svrmanager_addr are not mandatory anymore. So if someone starting proxy in prod forgets to initialize those, it will randomly crash at some point in the future when a request is made, which is not good.

Ideally we'd add eager defensive logic in eigenda_client initialization that errors in constructor if the svcmanager or ethrpc are uninitialized.

Reply:
doesn't this check already capture this? https://github.com/Layr-Labs/eigenda-proxy/blob/main/server/config.go#L48-L60

Reply:
Ah! But I think it's wrong now because even with cert enabled we still mandatorily need eth_rpc and svc_manager_addr in eigenda_client now right?

Reply:
isn't this the same check implicitly since they're both ingested via the same env source?

Reply:
Yes but my point is that the case: memstore off, cert verification off, still needs eth_rpc and svc_manager_addr to be populated. The check you shared above only runs when cert verification is on.

Reply:
ahh gotcha - will update the conditional

Reply:
[68a6979](https://github.com/Layr-Labs/eigenda-proxy/pull/205/commits/68a6979891597ebf4c470fade4e348aa4798f5e7)